lib/finite_machine/hooks.rb

```diff
@@ -18,7 +18,7 @@ class Hooks
     def initialize
       @hooks_map = Concurrent::Map.new do |events_hash, hook_event|
         events_hash[hook_event] = Concurrent::Map.new do |state_hash, name|
-          state_hash[name] = []
+          state_hash.compute_if_absent(name) { [] }
```

Comment:
If we're making things safer, it would probably make sense to convert this array to `Concurrent::Array.new`?